cmake/dependencies.cmake

```diff
@@ -71,7 +71,7 @@ unset(_ov_download_tbb_done CACHE)
 # or ENABLE_SYSTEM_TBB is OFF
 #
 function(ov_download_tbb)
-    if(_ov_download_tbb_done OR NOT THREADING MATCHES "^(TBB|TBB_AUTO)$")
+    if(_ov_download_tbb_done OR NOT THREADING MATCHES "^(TBB|TBB_AUTO|TBB_PARTITIONER_AUTO)$")
```

Comment:
Should we update the docs as well?
The name is a bit confusing.
What about 'TBB_ADAPTIVE' since 'TBB'_AUTO' is already taken.

Reply:
Done.